### PR TITLE
ENH: add kw `levels` to `Styler.hide_index` to hide only specific levels

### LINKED ISSUE #25475: New Feature Request: df.style.hide_index() to hide a particular index level
#### Problem description

The present implementation of `df.style.hide_index()` removes the complete index while rendering html. 
It would be better to have an optional input of index level and hide only that index level while rendering.
example  `df.style.hide_index(level=1)`
It provides more option and flexibility to user. 

 Sorry to post this as an issue, but I didn't find any option for suggesting new features.

### PR DESCRIPTION
**Pull request by attack68**
- [x] closes #25475
- [x] tests added / passed
- [x] whatsnew entry 1.4.0

Screen Shot 2021-07-21 at 14 13 35

### CODE REVIEW

File: pandas/io/formats/style.py [hunk 1]
```diff
@@ -1683,14 +1685,18 @@ def set_na_rep(self, na_rep: str) -> StylerRenderer:
         self.na_rep = na_rep
         return self.format(na_rep=na_rep, precision=self.precision)
 
-    def hide_index(self, subset: Subset | None = None) -> Styler:
+    def hide_index(
+        self,
+        subset: Subset | None = None,
+        levels: int | str | list[int | str] | None = None,
```
Comment (jreback) [inline pandas/io/formats/style.py:1691]: for followon, I believe we have Level to describe this typing.

File: pandas/io/formats/style.py [hunk 2]
```diff
@@ -1702,6 +1708,9 @@ def hide_index(self, subset: Subset | None = None) -> Styler:
             A valid 1d input or single key along the index axis within
             `DataFrame.loc[<subset>, :]`, to limit ``data`` to *before* applying
             the function.
+        levels : int, str, list
```
Comment (jreback) [inline pandas/io/formats/style.py:1711]: add a versionadded 1.4 here

**Review by jreback — CHANGES_REQUESTED** [review]
actually a couple of small comments, ping on green.

File: pandas/io/formats/style.py (hunk 1 above)
Comment (attack68) [reply pandas/io/formats/style.py]: changed, please review.

**Comment by attack68** [comment]
@jreback pinging on green

File: pandas/io/formats/style.py (hunk 2 above)
Comment (jreback) [reply pandas/io/formats/style.py:1775]: so we call this `level` (no s) elsewhere I think. can you update.

**Review by jreback — CHANGES_REQUESTED** [review]
pls rebase as well

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
failures unrelated

**Pull request merged**